File: src/com/example/javas/easy/E009_ReverseArray.java
/*
EASY PROBLEM (Simplified for 15 min):
Reverse an array in-place

REQUIREMENTS:
- Reverse the given array without creating a new array
- Modify the original array

APPROACH:
- Use two pointers: one at start, one at end
- Swap elements and move pointers towards center
- Stop when pointers meet

TIME COMPLEXITY: O(n)
SPACE COMPLEXITY: O(1)
*/

package com.example.javas.easy;

import java.util.Arrays;

public class E009_ReverseArray {
    
    public static void reverseArray(int[] arr) {
        // TODO: Implement array reversal
        // 1. Initialize left = 0, right = arr.length - 1
        // 2. While left < right:
        //    - Swap arr[left] and arr[right]
        //    - Increment left, decrement right
        
    }
    
    // TEST CASES
    public static void main(String[] args) {
        int[] arr1 = {1, 2, 3, 4, 5};
        reverseArray(arr1);
        System.out.println("Reversed [1,2,3,4,5]: " + Arrays.toString(arr1));
        
        int[] arr2 = {10, 20};
        reverseArray(arr2);
        System.out.println("Reversed [10,20]: " + Arrays.toString(arr2));
        
        int[] arr3 = {7};
        reverseArray(arr3);
        System.out.println("Reversed [7]: " + Arrays.toString(arr3));
        
        int[] arr4 = {3, 6, 9, 12};
        reverseArray(arr4);
        System.out.println("Reversed [3,6,9,12]: " + Arrays.toString(arr4));
    }
}

File: pythonX/easy/e009_reverse_array.py
"""
E009: Reverse Array

Summary:
Reverse an array in-place.

Requirements:
- Reverse the given array without creating a new array
- Modify the original array

Approach:
- Use two pointers: one at start, one at end
- Swap elements and move pointers towards center
- Stop when pointers meet

Big O Analysis:
Time Complexity: O(n)
Space Complexity: O(1)
"""

def reverse_array(arr: list[int]) -> None:
    """
    TODO: Implement array reversal
    1. Initialize left = 0, right = len(arr) - 1
    2. While left < right:
       - Swap arr[left] and arr[right]
       - Increment left, decrement right
    """
    pass


if __name__ == "__main__":
    # Test Cases
    arr1 = [1, 2, 3, 4, 5]
    reverse_array(arr1)
    print(f"Reversed [1,2,3,4,5]: {arr1}")
    
    arr2 = [10, 20]
    reverse_array(arr2)
    print(f"Reversed [10,20]: {arr2}")
    
    arr3 = [7]
    reverse_array(arr3)
    print(f"Reversed [7]: {arr3}")
    
    arr4 = [3, 6, 9, 12]
    reverse_array(arr4)
    print(f"Reversed [3,6,9,12]: {arr4}")
